Library/Homebrew/keg.rb

```diff
@@ -208,6 +208,7 @@ def ==(other)
 
   def empty_installation?
     Pathname.glob("#{path}/**/*") do |file|
+      return false if file.executable?
       next if file.directory?
       basename = file.basename.to_s
       next if Metafiles.copy?(basename)
```

Comment:
I think this check (and perhaps also for `INSTALL_RECEIPT.json` below but not `.DS_Store`) could instead be adjusted to look only in the `prefix` rather than in all subdirectories instead of checking whether things are executable. The reasoning for this is that some software doesn't set permissions properly so files may be set as executable unintentionally and subvert this check.